### PR TITLE
Pin release and RHCOS for 4.2 rebase

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by stbenjam** [comment]
I'll update the release image for KNI once https://github.com/openshift-metal3/dev-scripts/pull/577 is merged, for now it's pointing at openshift.

**Comment by russellb** [comment]
This is the last PR needed to resolve https://github.com/openshift-metal3/dev-scripts/issues/548 \o/

To get worker deployments to work with this PR, you'll need to add one more change.  Update `08_deploy_bmo.sh` to remove the `sed` command that switches to the `metalkube` branch of the `baremetal-operator`.

**Comment by stbenjam** [comment]
I've built the release image, and kicked off a test locally.

CI only tests kni-installer from git, I wonder if we should change it so it's only git for a kni-installer PR.  @derekhiggins Is there an env variable that I could check that has the PR url or repo under test?

**Comment by derekhiggins** [comment]
Build ABORTED, see build http://10.8.144.11:8080/job/dev-tools/691/

**Comment by stbenjam** [comment]
Restarted here http://10.8.144.11:8080/job/dev-tools/692/ - I needed to put @russellb's changes back for BMO

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/692/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/693/

**Comment by stbenjam** [comment]
CI is having some issue and fails early - is there stale data in .cache that didn't get cleaned out?

```
May 31 15:34:01 |  level=fatal msg="failed to fetch Terraform Variables: failed to generate asset \"Terraform Variables\": failed to get baremetal Terraform variables: failed to use cached libvirt image: open /home/notstack/.cache/kni-install/libvirt/image/3035b94eaf48da9850db0253308949a0.tmp: file exists"
```

**Comment by stbenjam** [comment]
Locally, the cluster comes up including worker:

```
NAME       STATUS   ROLES    AGE   VERSION
master-0   Ready    master   54m   v1.13.4+cb455d664
master-1   Ready    master   54m   v1.13.4+cb455d664
master-2   Ready    master   54m   v1.13.4+cb455d664
worker-0   Ready    worker   46s   v1.13.4+cb455d664
```

I've asked for 4.2.0-0.ci-2019-05-29-154816 to be pinned so it won't get garbage collected, we probably want that before merging this.  The relevant people seem to be out of office yesterday and today.

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/698/

**Comment by hardys** [comment]
Sorry, needs another rebase because I merged #588 but testing locally with that conflict resolved.

**Comment by hardys** [comment]
Ok I tested this with kni-installer building from source and it works fine :)

```
$ oc get nodes
NAME       STATUS   ROLES    AGE   VERSION
master-0   Ready    master   95m   v1.13.4+cb455d664
master-1   Ready    master   95m   v1.13.4+cb455d664
master-2   Ready    master   95m   v1.13.4+cb455d664
worker-0   Ready    worker   61m   v1.13.4+cb455d664

$ oc get clusterversion
NAME      VERSION                        AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.2.0-0.ci-2019-06-03-105723   True        False         84m     Cluster version is 4.2.0-0.ci-2019-06-03-105723
```

So this can be rebased and merged as soon as you're ready and have rebuilt the new kni-installer image

File: ocp_install_env.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ export EXTERNAL_SUBNET="192.168.111.0/24"
 # The release we default to here is pinned and known to work with our current
 # version of kni-installer.
 #
-export OPENSHIFT_RELEASE_IMAGE="${OPENSHIFT_RELEASE_IMAGE:-registry.svc.ci.openshift.org/kni/release:4.1.0-rc.5-kni.1}"
+export OPENSHIFT_RELEASE_IMAGE="${OPENSHIFT_RELEASE_IMAGE:-registry.svc.ci.openshift.org/ocp/release:4.2.0-0.ci-2019-06-03-105723}"
```
Comment (russellb) [inline ocp_install_env.sh]: shouldn't this be a KNI release payload?

Comment (stbenjam) [reply ocp_install_env.sh]: It will be

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/700/

**Comment by stbenjam** [comment]
This is good to go

```
[root@dell-r730-021 dev-scripts]# oc get nodes
NAME       STATUS   ROLES    AGE   VERSION
master-0   Ready    master   35m   v1.13.4+cb455d664
master-1   Ready    master   35m   v1.13.4+cb455d664
master-2   Ready    master   35m   v1.13.4+cb455d664
worker-0   Ready    worker   84s   v1.13.4+cb455d664
[root@dell-r730-021 dev-scripts]# oc get clusterversion
NAME      VERSION                              AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.2.0-0.ci-2019-06-03-105723-kni.0   True        False         18m     Cluster version is 4.2.0-0.ci-2019-06-03-105723-kni.0
```

**Pull request merged**